tests/perception/developmental_primitive_perception.py

```diff
@@ -0,0 +1,65 @@
+from adam.ontology.phase1_ontology import GAILA_PHASE_1_ONTOLOGY, MOM, DAD
+from adam.perception import PerceptualRepresentation
+from adam.perception.developmental_primitive_perception import (
+    DevelopmentalPrimitivePerceptionFrame,
+    ObjectPerception,
+    SENTIENT,
+    HasFlagProperty,
+    RgbColorPerception,
+    HasColor,
+    SUPPORTS,
+    RelationPerception,
+    CONTACTS,
+    BELOW,
+    ABOVE,
+    IsRecognizedParticular,
+)
+
+
+def test_recognized_particular():
+    # create a simple situation consisting of only Mom and Dad
+    mom = ObjectPerception("mom")
+    dad = ObjectPerception("dad")
+
+    PerceptualRepresentation.single_frame(
+        DevelopmentalPrimitivePerceptionFrame(
+            perceived_objects=[mom, dad],
+            property_assertions=[
+                HasFlagProperty(mom, SENTIENT),
+                HasFlagProperty(dad, SENTIENT),
+                IsRecognizedParticular(mom, MOM, ontology=GAILA_PHASE_1_ONTOLOGY),
+                IsRecognizedParticular(dad, DAD, ontology=GAILA_PHASE_1_ONTOLOGY),
+            ],
+        )
+    )
+
+
+def test_color():
+    # create a situation with a red ball
+    red = RgbColorPerception(255, 0, 0)
+    ball = ObjectPerception("ball")
+
+    PerceptualRepresentation.single_frame(
+        DevelopmentalPrimitivePerceptionFrame(
+            perceived_objects=[ball], property_assertions=[HasColor(ball, red)]
```

Comment:
As PyCharm warns I believe the arguments are supposed to be ImmutableSets, so I think we need to convert the lists with ImmutableSet(). The same is true for the arguments in the above test function.

Reply:
This is a PyCharm bug - it does not properly handle the `converter` applies by `attrs`.